tests/gnrc_tcp/tests/05-garbage-pkts.py

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/env python3
+
+# Copyright (C) 2019 Freie Universität Berlin
+#
+# This file is subject to the terms and conditions of the GNU Lesser
+# General Public License v2.1. See the file LICENSE in the top level
+# directory for more details.
+
+import base64
+import os
+import socket
+import sys
+import time
+
+from scapy.all import Ether, IPv6, TCP, raw, \
+                      sendp
+from testrunner import run
+
+from shared_func import sudo_guard, get_host_tap_device, get_host_ll_addr, \
+                        get_riot_if_id, get_riot_l2_addr, get_riot_ll_addr, \
+                        generate_port_number, verify_pktbuf_empty
+
+def testfunc(func):
+    def runner(child):
+        tap = get_host_tap_device()
+        host_ll = get_host_ll_addr(tap)
+        dst_if = get_riot_if_id(child)
+        dst_ll = get_riot_ll_addr(child)
+        dst_l2 = get_riot_l2_addr(child)
+        port = 33400 #generate_port_number()
+
+        # Setup RIOT Node wait for incoming connections from host system
+        child.sendline('gnrc_tcp_tcb_init')
+        child.expect_exact('gnrc_tcp_tcb_init: argc=1, argv[0] = gnrc_tcp_tcb_init')
+        child.sendline('gnrc_tcp_open_passive AF_INET6 {}'.format(port))
+        child.expect(r'gnrc_tcp_open_passive: argc=3, '
+                     r'argv\[0\] = gnrc_tcp_open_passive, '
+                     r'argv\[1\] = AF_INET6, argv\[2\] = (\d+)')
+        assert int(child.match.group(1)) == port
+
+        try:
+            print("- {} ".format(func.__name__), end="")
+            if child.logfile == sys.stdout:
+                func(child, tap, host_ll, dst_if, dst_l2, dst_ll, port)
+                print("")
+            else:
+                try:
+                    func(child, tap, host_ll, dst_if, dst_l2, dst_ll, port)
+                    print("SUCCESS")
+                except Exception as e:
+                    print("FAILED")
+                    raise e
+        finally:
+            child.sendline('gnrc_tcp_close')
+
+    return runner
+
+
+@testfunc
+def test_short_payload(child, src_if, src_ll, dst_if, dst_l2, dst_ll, dst_port):
+    # see https://github.com/RIOT-OS/RIOT/issues/11999
+    with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as sock:
+        sock.settimeout(child.timeout)
+        addr_info = socket.getaddrinfo(dst_ll + '%' + src_if, dst_port,
+                                       type=socket.SOCK_STREAM)
+        sock.connect(addr_info[0][-1])
+        child.expect_exact('gnrc_tcp_open_passive: returns 0')
+        child.sendline('gnrc_tcp_recv 1000000 1')
+        child.expect_exact('gnrc_tcp_recv: argc=3, '
```

Comment:
child.expect_exact line not really required here.

Reply:
See #12369 (comment), IMHO it is